### ISSUE TITLE
Enhancement: token.immediate(seq(someRegex, ...))

### ISSUE THREAD

**Issue by kylegoetz** [tree-sitter/tree-sitter#2351]
It would be useful to allow `token.immediate` to wrap a sequence that begins with a string or regex to make the parsing code simpler for nodes that can have an optional prefix:

```javascript
{
  prefix: _ => /[a-z]+/,
  suffix: _ => /[0-9]+/,
  imm_suffix: _ => token.immediate(/[0-9]+/)
  post_suffix: _ => token.immediate("postsuffix"),
  _tail: $ => seq($.suffix, $.post_suffix),
  _imm_tail: $ => seq($.imm_suffix, $.post_suffix),

  _term: $ => choice(
    seq($.prefix, optional($.imm_tail)),
    $._tail,
  )
}
```

could instead become

```javascript
{
  prefix: _ => /[a-z]+/,
  suffix: _ => /[0-9]+/,
  post_suffix: _ => token.immediate("postsuffix"),
  _tail: $ => seq($.suffix, $.post_suffix),

  _term: $ => choice(
    seq($.prefix, optional(token.immediate($._tail)),
    $._tail,
  )
}
```

We're able to get rid of some of the duplicative tokens where the only difference is one is immediate and the other is not. Just have one, and then when it msut be immediate, just wrap the whole thing in token.immediate (or phrase.immediate) or something.

Basically the behavior here would be that `tree-sitter generate` turns `phrase.immediate(seq(a1, a2, ..., an))` into `seq(token.immediate(a1), a2, . . . , an)`

**Comment by ObserverOfTime** [comment]
Duplicate of #931.
